Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -2210,6 +2314,80 @@ def fetch_channels_command():
     demisto.results("Successfully updated channels to the Integration Context")
 
 
+def fetch_users_from_api(user_name: str = ''):
+    """
+    Fetches users based on their name
+
+    Args:
+        user_name: Optional - The human readable form of a user's name.
+
+    Returns:
+        A list of dictionaries containing a user's name, ID, and email.
+    """
+    demisto.debug('Fetching users')
+    cursor = None
+    updated_users_list: list = []
+
+    res = CLIENT.users_list(limit=PAGINATED_COUNT, cursor=cursor)
+
+    while True:
+        cursor = res.get('response_metadata', {}).get('next_cursor')
+        members = res['members'] if res and res.get('members') else []
+        if user_name:
+            user_to_search = user_name.lower()
+            users_filter = list(filter(lambda u: u.get('name', '').lower() == user_to_search
+                                                 or u.get('profile', {}).get('email', '').lower() == user_to_search
+                                                 or u.get('real_name', '').lower() == user_to_search, members))
+            if users_filter:
+                return users_filter[0]
+        else:
+            for member in members:
+                updated_member: dict = {
+                    'name': member.get('name'),
+                    'id': member.get('id'),
+                    'real_name': member.get('real_name', ''),
+                    'profile': {
+                        'email': member.get('email', '')
+                    }
+                }
+                updated_users_list.append(updated_member)
+        if not cursor:
+            demisto.info("Reached the end of looking for a user")
+            break
+        else:
+            total_try_time = 0
+            while True:
+                try:
+                    res = CLIENT.users_list(limit=PAGINATED_COUNT, cursor=cursor)
+
+                except SlackApiError as api_error:
+                    demisto.debug(f'Got rate limit error (sync). Body is: {str(res)}\n{api_error}')
+                    response = api_error.response
+                    headers = response.headers  # type: ignore
+                    if 'Retry-After' in headers:
+                        retry_after = int(headers['Retry-After'])
+                        total_try_time += retry_after
```

Comment:
you should increment total_try_time before `if 'Retry-After' in headers` because now there is non increment inn case  'Retry-After' not in headers.